usaspending_api/accounts/v2/filters/account_download.py

```diff
@@ -159,64 +160,75 @@ def get_nonzero_filter():
+        q = _build_submission_queryset(selected_period)
+    else:
+        closed_periods = get_last_closed_periods_per_year()
+        for closed_period in closed_periods:
+            q |= _build_submission_queryset(closed_period)
+
+    return q
+
 
-        submission_filter = Case(
-            When(q, then=F(column_name)), default=Cast(Value(None), DecimalField(max_digits=23, decimal_places=2))
+def _build_submission_queryset_for_derived_fields(submission_closed_period_queryset, column_name):
+    if submission_closed_period_queryset:
+        queryset = Case(
+            When(submission_closed_period_queryset, then=F(column_name)),
+            default=Cast(Value(None), DecimalField(max_digits=23, decimal_places=2)),
         )
     else:
-        submission_filter = Cast(Value(None), DecimalField(max_digits=23, decimal_places=2))
-
-    return submission_filter
+        queryset = F(column_name)
+    return queryset
 
 
-def generate_ussgl487200_derived_field(filters):
+def generate_ussgl487200_derived_field(submission_queryset=None):
     column_name = "ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe"
-    return _generate_closed_period_for_derived_field(filters, column_name)
+    return _build_submission_queryset_for_derived_fields(submission_queryset, column_name)
 
 
-def generate_ussgl497200_derived_field(filters):
+def generate_ussgl497200_derived_field(submission_queryset=None):
     column_name = "ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe"
-    return _generate_closed_period_for_derived_field(filters, column_name)
+    return _build_submission_queryset_for_derived_fields(submission_queryset, column_name)
 
 
-def generate_gross_outlay_amount_derived_field(filters, account_type):
+def generate_gross_outlay_amount_derived_field(account_type, submission_queryset=None):
     column_name = {
         "account_balances": "gross_outlay_amount_by_tas_cpe",
         "object_class_program_activity": "gross_outlay_amount_by_program_object_class_cpe",
         "award_financial": "gross_outlay_amount_by_award_cpe",
     }[account_type]
 
-    return _generate_closed_period_for_derived_field(filters, column_name)
+    return _build_submission_queryset_for_derived_fields(submission_queryset, column_name)
 
 
-def generate_treasury_account_query(queryset, account_type, filters):
+def generate_treasury_account_query(queryset, account_type):
     """ Derive necessary fields for a treasury account-grouped query """
     derived_fields = {
         "submission_period": get_fyp_or_q_notation("submission"),
-        "gross_outlay_amount": generate_gross_outlay_amount_derived_field(filters, account_type),
-        "gross_outlay_amount_fyb_to_period_end": generate_gross_outlay_amount_derived_field(filters, account_type),
+        "gross_outlay_amount": generate_gross_outlay_amount_derived_field(account_type),
+        "gross_outlay_amount_fyb_to_period_end": generate_gross_outlay_amount_derived_field(account_type),
```

Comment:
Will `gross_outlay_amount` and `gross_outlay_amount_fyb_to_period_end` always be the same because `generate_gross_outlay_amount_derived_field` is called for both with the same parameter? Same for `generate_federal_account_query`

Reply:
The `account_balances` download refers to `gross_outlay_amount` and the other two refer to `gross_outlay_amount_fyb_to_period_end`. Both are annotated and then the correct columns are used for each download.

Reply:
Oh, I see what you mean. Gotcha